.github/copilot-instructions.md

```diff
@@ -0,0 +1,171 @@
+  - Include resource tagging for cost tracking
+
+- **TypeScript/JavaScript**:
+  - Follow standard ESLint configuration
+
+- **YAML**:
+  - Use consistent indentation (2 spaces)
+  - Follow Porter best practices for bundle manifests
+
+- **Git**:
+  - Update CHANGELOG.md for all significant changes
+  - Reference issue numbers in commit messages
+
+## Environment Assumptions
+
+- **Azure**: All resources are deployed in Azure
+- **Microsoft Entra ID**: Used for authentication and authorization
+- **Networking**: Core infrastructure uses hub-spoke networking model
+- **Security**: Zero-trust security model with strict network boundaries
+- **Deployment**: CI/CD through GitHub Actions
+
+## Template Structure
+
+Azure TRE uses Porter bundles to define workspaces, workspace services, and user resources. These bundles consist of:
+
+### porter.yaml
+
+This is the main Porter manifest file that defines:
+- Credentials required for deployment
+- Parameters and their defaults
+- Actions (install, upgrade, uninstall)
+- Mixins used (terraform, exec, etc.)
+- Outputs from the deployment
+
+Example structure:
+```yaml
+name: tre-service-example
+version: 0.1.0
+description: "An example TRE service"
+registry: azuretre
+...
+```
+
+### template_schema.json
+
+JSON Schema file that defines the parameters that can be provided when deploying a resource. It follows standard JSON Schema format and is used by the API and UI to generate forms for resource creation.
+
+Key sections include:
+- Properties with types, descriptions, and defaults
+- Required fields
+- UI schema for customizing the display in the UI
+
+### Terraform
+
+Most bundles use Terraform to provision the actual resources in Azure. Common practices include:
+- Breaking Terraform code into modules
+- Using remote state for complex deployments
+- Applying proper tagging for cost tracking
+- Using variables.tf, main.tf, and outputs.tf
+- Including lifecycle blocks to prevent resource recreation
+
+## Changelog Updates
+
+When creating a pull request, you must update the CHANGELOG.md file with your changes. Add your changes under one of these sections in the unreleased section at the top:
+
+- ENHANCEMENTS
+- BUG FIXES
+- COMPONENTS (for version updates)
+
+Format for changelog entries:
+```markdown
+* Brief description of change ([#1234](https://github.com/microsoft/AzureTRE/issues/1234))
+```
+
+Always include issue and/or PR references using the format `([#1234](https://github.com/microsoft/AzureTRE/issues/1234))`.
+
+## Version Management
+
+When editing components or bundles, you must increase their version numbers according to semantic versioning principles:
+
+1. **MAJOR** version: Breaking changes, potential data loss, significant changes requiring review
+2. **MINOR** version: New functionality with automatic upgrade capability
+3. **PATCH** version: Backward-compatible bug or typo fixes
+
+Version updates should be documented in:
+1. The component's version file (porter.yaml for bundles)
```

Comment:
also add locations of version for api, airlock processor, ui and resource processor.